### DISCUSSION TITLE
Debugging vscode-cpptools / Extending vscode-cpptools

### DISCUSSION THREAD

**Discussion by theIDinside** [microsoft/vscode-cpptools#8333] [Q&A]
So I've got a few ideas that I need to implement, that involves the debugging extension (cppdbg) - and in order to learn the codebase and understand how it works, I would like to debug it - generally I think one can just do the regular vscode extension debugging; namely hit ```F5``` and then from an extension host do what I need to do, but this doesn't seem to work.

Are there any guides for how one would contribute features and/or extend, or possibly even fork (since the features I am talking about is pretty niche) cppdbg? Must I uninstall my global installation of the extension in order to debug the git cloned one?

When I hit ```F5```, I get this error from the extension host window;

```
Activating extension 'ms-vscode.cpptools' failed: Cannot find module '<path_to...>/vscode-cpptools/Extension/dist/main'
Require stack:
- /usr/share/code/resources/app/out/vs/loader.js
- /usr/share/code/resources/app/out/bootstrap-amd.js
- /usr/share/code/resources/app/out/bootstrap-fork.js.
```
Replaced the path. But the rest of the message is "as is" so to speak.

**Comment by sean-mcmanus** [comment]
See https://github.com/microsoft/vscode-cpptools/blob/main/CONTRIBUTING.md and https://github.com/microsoft/vscode-cpptools/blob/main/Documentation/Building%20the%20Extension.md .

**Reply by wangzhankun** [reply]
I am ready with the pre-requisites: 1. checkout to `release` branch; 2. run `yarn install`
<img width="269" alt="image" src="https://github.com/microsoft/vscode-cpptools/assets/35724924/6e28ade8-1c3e-4f49-ac90-26e49fdc7f17">
But after I press `F5`, I meet the same issue:
```log
Activating extension 'ms-vscode.cpptools' failed: Cannot find module '/home/wang/Documents/github/vscode-cpptools/Extension/dist/main' Require stack: - /usr/share/code/resources/app/out/vs/loader.js - /usr/share/code/resources/app/out/bootstrap-amd.js - /usr/share/code/resources/app/out/bootstrap-fork.js - .
```
How to solve it?
